src/sentry/models/group.py

```diff
@@ -369,6 +369,41 @@ def by_qualified_short_id_bulk(
             ).filter(short_id_lookup, project__organization=organization_id)
         )
         group_lookup: set[int] = {group.short_id for group in groups}
+
+        # If any requested short_ids are missing after the exact slug match,
+        # fallback to a case-insensitive slug lookup to handle legacy/mixed-case slugs.
+        missing_short_ids = [sid.short_id for sid in short_ids if sid.short_id not in group_lookup]
+        if missing_short_ids:
+            # Build a lookup only for the missing short_ids per slug
+            missing_by_slug = defaultdict(list)
+            for sid in short_ids:
+                if sid.short_id in missing_short_ids:
+                    missing_by_slug[sid.project_slug].append(sid.short_id)
+
+            ci_short_id_lookup = reduce(
+                or_,
+                [
+                    Q(project__slug__iexact=slug, short_id__in=sids)
+                    for slug, sids in missing_by_slug.items()
+                ],
+            )
+
+            fallback_groups = list(
+                self.exclude(
+                    status__in=[
+                        GroupStatus.PENDING_DELETION,
+                        GroupStatus.DELETION_IN_PROGRESS,
+                        GroupStatus.PENDING_MERGE,
+                    ]
+                ).filter(ci_short_id_lookup, project__organization=organization_id)
+            )
+
+            if fallback_groups:
+                # Merge unique groups from fallback
+                existing_ids = {g.id for g in groups}
+                groups.extend(g for g in fallback_groups if g.id not in existing_ids)
+                group_lookup = {group.short_id for group in groups}
```

Comment:
isn't there guaranteed no overlap between fallback_groups and groups? do we need the additional check?
```suggestion
                groups.extend(g for g in fallback_groups)
                group_lookup = {group.short_id for group in groups}
```